include/aspect/mesh_refinement/isotherms.h

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright (C) 2011, 2012 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+/*  $Id: positiontime.h 1433 2012-12-08 08:24:55Z bangerth $  */
+
+
+#ifndef __aspect__mesh_refinement_isotherms_h
+#define __aspect__mesh_refinement_isotherms_h
+
+#include <aspect/mesh_refinement/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+
+    /**
+     * A class that implements a mesh refinement criterion based on
+     * isotherms.
+     *
+     * @ingroup MeshRefinement
+     */
+    template <int dim>
+    class Isotherms : public Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * After cells have been marked for coarsening/refinement, apply
+         * additional criteria independent of the error estimate.
+         *
+         */
+        virtual
+        void
+        tag_additional_cells () const;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+     private:
+        /**
+         * Declare a variable which indicates if and which compositional field should be excluded from the process.
+         */
+        signed int exclude_composition; 
+        /**
+         * Declare a variable which indicates how many isotherms the user has declared, and a variable to store all those isotherms
+         */
+        int number_of_isotherms;
+	std::vector<std::vector<std::string> > isotherms;
```

Comment:
This is duplicative. You can always get the number of isotherms via `isotherms.size()` and this is preferable to storing the same information twice (because invariable these variables will eventually get out of synch as code is being developed further).